.evg.yml

```diff
@@ -3,100 +3,196 @@ functions:
+            fi
+          done
+  "publish_codecov":
+    - command: shell.exec
+      params:
+        shell: "bash"
+        silent: true
+        script: |
+          set -e
+          echo export CODECOV_TOKEN=${codecov_token} >> creds
+    - command: shell.exec
+      params:
+        shell: "bash"
+        script: |
+          source ./creds
+          cd stitch-ios-sdk
+
+          if [ -z "${github_pr_number}" ]
+          then
+            if [ "${is_patch}" = "true" ]; then
+              exit 0
+            fi
+            bash <(curl -s https://codecov.io/bash) -C ${revision} -b ${build_id}
+          else
+            PR_INFO=`curl https://api.github.com/repos/${github_org}/${github_repo}/pulls/${github_pr_number}`
+            BRANCH=`echo $PR_INFO | python -c "import sys, json; print json.load(sys.stdin)['head']['ref']"`
+            bash <(curl -s https://codecov.io/bash) -B $BRANCH -C ${revision} -P ${github_pr_number} -b ${build_id}
+          fi
 tasks:
-  - name: github_pull_request_start
-    priority: 100
+  - name: lint
     commands:
+      - func: "fetch_source"
       - command: shell.exec
-        params:
+        params: 
+          shell: "bash"
           script: |
-            if ! [ "${is_patch}" = "true" ]; then
-              exit 0
-            fi
-            echo pending > commit_state
-      - func: "set_github_build_status"
-
-  - name: github_pull_request_end
-    depends_on:
-      - name: github_pull_request_start
-      - name: run_tests
-        status: "*"
-        variant: linux-64
+            set -e
+            cd stitch-ios-sdk/StitchCore
+            swiftlint
+  - name: run_core_tests
     commands:
+      - func: "fetch_source"
+      - func: "setup_xcov"
+      - func: "setup_mongod"
+      - func: "setup_stitch"
       - command: shell.exec
-        params:
-          silent: true
+        params: 
+          shell: "bash"
           script: |
-            if ! [ "${is_patch}" = "true" ]; then
-              exit 0
-            fi
-            BUILDS=( `curl -H "Auth-Username: ${evg_rest_user}" -H "Api-Key: ${evg_rest_api_key}"  https://evergreen.mongodb.org/rest/v1/versions/${version_id} | python -c "import sys, json; builds = json.load(sys.stdin)['builds']; print ' '.join(builds)"` )
-            for build in "${BUILDS[@]}"
-            do
-              curl -H "Auth-Username: ${evg_rest_user}" -H "Api-Key: ${evg_rest_api_key}"  https://evergreen.mongodb.org/rest/v1/builds/$build | python -c "import sys, json; tasks = json.load(sys.stdin)['tasks']; sys.exit(0) if all(tasks[task]['status'] == 'success' or task == 'github_pull_request_end'  for task in tasks) else sys.exit(1)"              
-              if [ $? -ne 0 ]; then
-                echo "failure" > commit_state
-                exit 0
-              fi
-            done
-
-            echo "success" > commit_state
-      - func: "set_github_build_status"
+            set -e
+            cd stitch-ios-sdk/StitchCore
```

Comment:
[q] Can we run a script that tests everything, not just core? That's what java is doing